paper/main.tex

```diff
@@ -924,107 +938,99 @@ \section{Evaluation of performance}\label{section:evaluation_of_performance}
-
-Furthermore, in Section~\ref{section:top_performances}
-and~\ref{section:evaluation_of_performance} it was shown that envious strategies
-performed well. Though these were not the most envious strategies in the
-tournaments (ZDs were included), these strategies benefited by being a bit
-envious. From Section~\ref{section:evaluation_of_performance} it was concluded
-that there is a significant importance in adapting to the environment, and more
-specifically in this work, to the mean cooperator. This section also
-demonstrated that a strategy should reciprocate, as suggested by R. Axelrod, but in
-some environments, such as standard and noisy, it should relax its readiness to
-do so.
-
-Thus, the five properties successful strategies need to have in a IPD competition
-are:
+This manuscript explored the performance of \numberofstrategies strategies of
+the IPD in \numberofalltournaments computer tournaments. The collection of
+computer tournaments presented here is the largest and more diverse collection in the
+literature. The \numberofstrategies strategies are drawn from known named
+strategies in IPD literature and the computer tournaments include tournaments of
+four different settings.
+
+So what is the best way of playing the IPD? And is
+there a single dominant strategy for the IPD? There was not a single strategy
+within the collection of the \numberofstrategies strategies that managed to
+perform well in all the tournaments variations it competed in. Even if on
+average a strategy ranked highly in a specific environment this did not
+guarantee its success over the different tournament types. Nevertheless, in
+Sections~\ref{section:top_performances}
+and~\ref{section:evaluation_of_performance} we examined the best performing
+strategies across various tournament types and analysed their salient features,
+and demonstrated that there are properties associated with the success of
+strategies, contradicting the originally suggested properties of R.
+Axelrod~\cite{Axelrod1981}.
+
+We showed that complex or \textbf{clever} strategies can be effective, whether
+trained against a corpus of possible opponents or purposely designed to mitigate
+the impact of noise such as in the strategy DBS. Moreover, we found some
+strategies designed or trained for noisy environments were also highly ranked in
+noise-free tournaments which reinforces the idea that strategies'
+complexity/cleverness is not necessarily a liability, rather it can confer
+adaptability to a more diverse set of environments.
+We also showed that while the type of exploitation attempted by ZDs is
+not typically effective in standard tournaments, \textbf{envious} strategies
+capable of both exploiting and not their opponents can be highly successful.
+Based on the results of~\cite{Harper2017} this could be because they are
+selectively exploiting weaker opponents while mutually cooperating with stronger
+opponents. Highly noisy or tournaments with short matches also favoured envious
```

Comment:
I like this :+1:

Reply:
I agree. At the risk of giving a terrible analogy, biological creatures have some very complicated subsystems (such as the immune system) that confer generalized environmental adaptability, which requires remembering and exploiting historical information.